nixos/modules/installer/tools/nixos-rebuild.sh

```diff
@@ -422,11 +426,43 @@ if [ "$action" = dry-build ]; then
     extraBuildFlags+=(--dry-run)
 fi
 
+if [ "$action" = list-generations ]; then
+    generation_from_dir() {
+        generation_dir="$1"
+        generation="$(basename "$generation_dir")" # Has the format "system-123-link" for generation 123
+        generation="${generation#*-}"              # Delete everything before the first "-"
+        generation="${generation%%-*}"             # Delete everything after the last "-"
+        echo "$generation"
+    }
+    describe_generation(){
+        generation_dir="$1"
+        generation="$(generation_from_dir "$generation_dir")"
+        nixos_version="$(cat "$generation_dir/nixos-version" 2> /dev/null || echo "Unknown")"
+
+        kernel_dir="$(dirname "$(realpath "$generation_dir/kernel")")"
+        kernel_version="$(ls "$kernel_dir/lib/modules")"
+
+        build_date="$(date --date="@$(stat "$generation_dir" --format=%W)" "+%a %F %T")"
+
+        echo "$generation,$nixos_version,$kernel_version,$build_date"
+    }
+
+    description=""
+    for generation_dir in /nix/var/nix/profiles/system-*-link ; do
+        description="$description\n$(describe_generation "$generation_dir")"
```

Comment:
I don't particularly like this, but I don't know how to do this in a nicer way...

Reply:
you could use a bash array, I guess